corehq/apps/commtrack/sms.py

```diff
@@ -370,7 +371,7 @@ def single_action_transactions(self, action, args):
                         case_id=self.case_id,
                         product_id=p.get_id,
                         action=const.StockActions.RECEIPTS,
-                        quantity=value.split('.')[1]
+                        quantity=Decimal(value.split('.')[1])
```

Comment:
No I think it just wasn't being validated upfront, and that django's DecimalField accepts a string without complaining.